pages/contract/_contract.vue

```diff
@@ -1,138 +1,37 @@
+      :contractName="contractName"
+    />
+    <contract-tokens
+      :contractDetails="contractDetails"
+    />
   </v-container>
 </template>
 
 <script lang="ts">
-/* eslint-disable getter-return */
 import { Component, namespace, Vue } from "nuxt-property-decorator"
 import { LSPConfiguration } from "~/types"
 
-const contracts = namespace("contracts")
-
-const addresses: Record<string, string> = require("@/addresses.json")
-
-const readableLibraryNames: Record<string, string> = {
-  LinearLongShortPairFinancialProductLibrary: "Linear Payout Contract",
-}
-
-const libraryParameters: Record<string, Array<string>> = {
-  LinearLongShortPairFinancialProductLibrary: ["Upper Bound", "Lower Bound"],
-}
+import ContractSummary from "@/components/ContractSummary.vue"
+import ContractTokens from "@/components/ContractTokens.vue"
 
-const priceFeedLinks: Record<string, string> = {
-  USDETH: "https://github.com/UMAprotocol/UMIPs/blob/master/UMIPs/umip-6.md",
-}
+const contracts = namespace("contracts")
 
-@Component
-export default class Contract extends Vue {
+@Component({ components: { ContractSummary, ContractTokens } })
+export default class contract extends Vue {
   @contracts.State
   contractConfigs!: Array<LSPConfiguration>
 
-  get contractName(): string {
-    return this.$route.params.contract
-  }
-
-  get expirationTime(): string | void {
-    if (this.contractDetails) {
-      return new Date(this.contractDetails.expirationTime).toLocaleString()
-    }
-  }
-
-  get libraryConfiguration(): Array<object> | undefined {
-    if (this.contractDetails) {
-      return this.contractDetails.financialProductLibraryParameters.map(
-        (value, index) => {
-          if (this.contractDetails) {
-            return {
-              name:
-                libraryParameters[this.contractDetails.financialProductLibrary][
-                  index
-                ],
-              value,
-            }
-          } else {
-            return {}
-          }
-        }
-      )
-    }
-  }
-
-  get etherscanLinkContract(): string | undefined {
-    if (this.contractDetails) {
-      return `https://kovan.etherscan.io/address/${this.contractDetails.address}`
-    }
-  }
-
-  get etherscanLinkCollateral(): string | undefined {
-    if (this.contractDetails) {
-      return `https://kovan.etherscan.io/address/${
-        addresses[this.contractDetails.collateralToken]
-      }`
-    }
-  }
-
-  get priceFeedLink(): string | undefined {
-    if (this.contractDetails) {
-      return priceFeedLinks[this.contractDetails.priceIdentifier]
-    }
-  }
-
   get contractDetails(): LSPConfiguration | undefined {
     return this.contractConfigs.find(
       (config) => config.syntheticName === this.contractName
     )
   }
 
-  get contractType(): string | undefined {
-    if (this.contractDetails)
-      return readableLibraryNames[this.contractDetails.financialProductLibrary]
+  get contractName(): string {
+    return this.$route.params.contract
```

Comment:
I'm not having any issues with autocomplete on the this.$route object, are you still having an issue?

Reply:
I still get an error message. But this might very well just be due to misconfiguration on my side.
![image](https://user-images.githubusercontent.com/15629702/123421341-fff8de80-d5bc-11eb-8a59-0d4fa09759ad.png)

Reply:
whack. Open this as an issue so we take a look later on.